Tools/TestWebKitAPI/Tests/ios/DragAndDropTestsIOS.mm

```diff
@@ -2242,6 +2237,57 @@ static BOOL isCompletelyWhite(UIImage *image)
     EXPECT_WK_STREQ("com.pixar.universal-scene-description-mobile", [registeredTypes firstObject]);
 }
 
+#if ENABLE(MODEL_PROCESS)
+
+TEST(DragAndDropTests, CheckModelDragPreview)
+{
+    auto configuration = adoptNS([[WKWebViewConfiguration alloc] init]);
+    for (_WKFeature *feature in [WKPreferences _features]) {
+        if ([feature.key isEqualToString:@"ModelElementEnabled"] || [feature.key isEqualToString:@"ModelProcessEnabled"])
+            [[configuration preferences] _setEnabled:YES forFeature:feature];
+    }
+
+    auto messageHandler = adoptNS([[ModelLoadingMessageHandler alloc] init]);
+    [[configuration userContentController] addScriptMessageHandler:messageHandler.get() name:@"modelLoading"];
+
+    auto webView = adoptNS([[TestWKWebView alloc] initWithFrame:CGRectMake(0, 0, 320, 500) configuration:configuration.get()]);
+    [webView synchronouslyLoadTestPageNamed:@"simple-model-page"];
+
+    while (![messageHandler modelIsReady])
+        Util::spinRunLoop();
+
+    [webView waitForNextPresentationUpdate];
+
+    bool done = false;
+
+    auto simulator = adoptNS([[DragAndDropSimulator alloc] initWithWebView:webView.get()]);
+    [simulator setDragProgressMidPointReachedBlock:makeBlockPtr([&done, simulator, webView] () mutable {
+        EXPECT_TRUE([[simulator liftPreviews].firstObject.view window] != nil);
+
+        [webView evaluateJavaScript:@"document.querySelector('model').remove()" completionHandler:makeBlockPtr([&done, simulator, webView] (NSNumber *value, NSError *error) mutable {
+            [webView waitForNextPresentationUpdate];
+            NSArray *liftPreviews = [simulator liftPreviews];
+            UITargetedDragPreview *liftPreview = (UITargetedDragPreview *)liftPreviews.firstObject;
+            EXPECT_EQ(_UIRemoteView.class, liftPreview.view.class);
```

Comment:
Maybe `-isKindOfClass:` instead? (So that it's robust in case it gets its own subclass in the future)